nvm.sh

```diff
@@ -2360,6 +2392,49 @@ nvm_npmrc_bad_news_bears() {
   return 1
 }
 
+nvm_windows_drive_posix() {
+  nvm_grep -iEe "^$1 " /proc/mounts 2>/dev/null | command awk '{ print $2 }'
+}
+
+# args: path to check and transform if needed
+nvm_ensure_posix_path() {
+  if [ -z "${1-}" ]; then
+    nvm_err "Need the location as first parameter"
+    return 1
+  fi
+  local location
+  location="$1"
+
+  # If already posix path, just return it
+  if [ "$(command printf '%s' "$location" | command cut -b 1)" = "/" ]; then
+    nvm_echo "$location"
+    return 0
+  fi
+
+  # If absolute path (eg. starting with C:)
+  if command printf '%s' "$location" | command grep '^[a-zA-Z]:' >/dev/null 2>&1; then
+    # Get drive letter, eg. C:
+    local drive
+    drive=$(command printf '%s' "$location" | command cut -b -2)
+    local prefix
+    prefix=$(nvm_windows_drive_posix "$drive")
+
+    if [ -z "${prefix:-}" ]; then
+      nvm_err "Unable to find where '$drive' is mounted"
+      return 1
+    fi
+
+    # Remove drive letter prefix (eg. C:)
+    location=$(command printf '%s' "$location" | command cut -b 3-)
```

Comment:
can all these `printf` calls use `nvm_echo` instead?

Reply:
I think I faced issue for some of them because nvm_echo is printing a new line

Reply:
Or maybe it was just that I did not like the new line added by nvm_echo
Or probably because i updated on my other project i use it and obviously it does not have nvm_echo :D
I ll put nvm_echo then

Reply:
Of course, neither this nor the previous comments would apply if `nvm_ensure_posix_path` is dropped in favor of the cd/pwd solution...

Reply:
true; that seems cleaner

Reply:
It also eliminates the `nvm_windows_drive_posix` function and the need to unit-test it...  and speeds things up a bit for Cygwin and MSYS.

(Because those platforms do not support a true `fork`, every single external command or subshell takes a substantial fraction of a second.  Loading nvm and doing an initial `use` on Cygwin currently takes a full 12 seconds for me; `nvm ls` takes over 30 seconds.)